### PR TITLE
style(grid): add styles for boolean column

### PR DESCRIPTION
**Pull request by didimmova**
### Additional information (check all that apply):
 - [ ] Bug fix
 - [ ] New functionality
 - [ ] Documentation
 - [ ] Demos
 - [ ] CI/CD

### Checklist:
 - [x] All relevant tags have been applied to this PR
 - [ ] This PR includes unit tests covering all the new code ([test guidelines](https://github.com/IgniteUI/igniteui-angular/wiki/Test-implementation-guidelines-for-Ignite-UI-for-Angular))
 - [ ] This PR includes API docs for newly added methods/properties ([api docs guidelines](https://github.com/IgniteUI/igniteui-angular/wiki/Documentation-Guidelines))
 - [ ] This PR includes `feature/README.MD` updates for the feature docs
 - [ ] This PR includes general feature table updates in the root `README.MD`
 - [ ] This PR includes `CHANGELOG.MD` updates for newly added functionality
 - [ ] This PR contains breaking changes
 - [ ] This PR includes `ng update` migrations for the breaking changes ([migrations guidelines](https://github.com/IgniteUI/igniteui-angular/wiki/Update-Migrations))
 - [ ] This PR includes behavioral changes and the feature specification has been updated with them

### CODE REVIEW

File: projects/igniteui-angular/src/lib/core/styles/components/grid/_grid-theme.scss [hunk 1]
```diff
@@ -1139,6 +1139,25 @@
         bottom: 0;
     }
 
+    %igx-grid__td--bool {
+        .igx-grid__td-text {
```
Comment (simeonoff) [inline projects/igniteui-angular/src/lib/core/styles/components/grid/_grid-theme.scss:1143]: You should **never** refer to css class selectors generated using the BEM mixins here. Also, you don't have to modify the text field.

File: projects/igniteui-angular/src/lib/core/styles/components/grid/_grid-theme.scss [hunk 2]
```diff
@@ -1139,6 +1139,25 @@
...
+            margin: auto;
+        }
+
+        igx-icon {
+            height: rem(18px);
+            width: rem(18px);
+            font-size: rem(18px);
+            color: igx-color($palette, 'grays', 500);
+
+            &.igx-grid__td--bool-check {
```
Comment (simeonoff) [inline projects/igniteui-angular/src/lib/core/styles/components/grid/_grid-theme.scss:1155]: Same as above. Don't do this.

Comment (didimmova) [reply projects/igniteui-angular/src/lib/core/styles/components/grid/_grid-theme.scss:1155]: This one is not created by the BEM mixins, it'a custom.

Comment (simeonoff) [reply projects/igniteui-angular/src/lib/core/styles/components/grid/_grid-theme.scss:1155]: I can tell by the name as it doesn't follow the BEM naming convention. You should use BEM to generate the class names. Doing what you did here is generally not allowed, or only allowed in some edge cases.

File: projects/igniteui-angular/src/lib/core/styles/components/grid/_grid-theme.scss [hunk 3]
```diff
@@ -1139,6 +1139,26 @@
...
+
+        igx-icon {
+            height: rem(18px);
+            width: rem(18px);
+            font-size: rem(18px);
+            color: igx-color($palette, 'grays', 500);
+        }
+
+
+        &%igx-grid__td--bool-true {
```
Comment (simeonoff) [inline projects/igniteui-angular/src/lib/core/styles/components/grid/_grid-theme.scss:1153]: No need to nest this here.

**Comment by Aleksandyr** [comment]
There is a difference in the opacity of the icons. Is it intentional?
image

**Comment by didimmova** [comment]
Absolutely. It follows the spec.

**Comment by Aleksandyr** [comment]
@didimmova It seems that the spec says that there is a slight difference in the icons.

**Pull request merged**